Feedback implementation

dogoyaro opened the pull request.

dogoyaro committed:
refactor actions

dogoyaro committed:
refactor views to add phone number in registration

dogoyaro committed:
update socket connection with heroku url

Commit:
emit using user id as identifier

Commit:
clear groups after emitting

Commit:
use id sent from client side to create group for user

Commit:
format code, repair indentation

Commit:
add new build files

Commit:
modify group routes to store users in list and send groups for specific users

Commit:
send user id to add group action

Commit:
change socket.io connection parameter to heroku hosted app

Commit:
new build files

Commit:
add new build files

Commit:
add new babel-watch

Commit:
new temporary changes

Commit:
repair indentation

dogoyaro committed:
modify get groups route, remove user management, clear groups and groupkeys on .on callback

Commit:
Merge branch 'add-phone-number' of https://github.com/OsayamenOsaretin/post-it-app into add-phone-number

pull changes from remote
.

Commit:
clear group routes and maps in firebase on event listener

Commit:
remove babel-watch, use babel-node

Commit:
add user id to socket.io listener

Commit:
modify message store, send message response directly to message map

Commit:
use object.values to iterate over group ids

Commit:
reduce max len to 80

Commit:
refactor message body

Commit:
add new build files

Commit:
remove unwanted logs from message route

Commit:
use heroku url for socket.io connection

dogoyaro committed:
repair indentation

dogoyaro committed:
repair indentation

dogoyaro committed:
add comment to explain group and groupmap clear

Commit:
add empty line to end of file

Commit:
add new build files

Commit:
add user status change handler, update user status map

Commit:
correct heroku connection on dashboard

Commit:
change user name to group request sent in user list view

Commit:
add new constants for recieve requests

Commit:
add recieve request action

Commit:
change user to group route to requests

Commit:
temporary request functionality

Commit:
temporary request functionality

Commit:
add request routes

Commit:
add new build files

Commit:
temporary features and fixes

Commit:
add styles for group invites

Commit:
add tags for styling to components

Commit:
add new build files

Commit:
new coverage files

Commit:
refactor code style

Commit:
comment out unused code temporarily

Commit:
feedback-change from module.exports to export default

Commit:
feedback-change from module.exports to export default

Commit:
remove non mvc routes

Commit:
add controller function (implement feedback)

Commit:
seperate controllers into different files

Commit:
add route index

Commit:
add socket configuration

Commit:
add new group request style

Commit:
add dist folder to git ignore

Commit:
import function to export socket instance

Commit:
remove unneccessary log messages

Commit:
remove dist files from files to be committed

Commit:
add refactored test files

Commit:
add mocks for firebase and socket.io

Commit:
remove coverage files

Commit:
feedback(move folder to seperate folder)

Commit:
feedback(add empty line at the end of file)

Commit:
chore(add prod scripts): add scripts to run in production

Commit:
refactor(index.css): add style for priority level button

Commit:
chore(webpack.config.prod): add webpack config for production

Commit:
feedback(app.jsx): remove commented out code.

Commit:
refactor(move files to individual test folder)

Commit:
chore(firebase functions): add functions.

Commit:
refactor(implement feedback)

Commit:
refactor(remove unused container folder/file)

Commit:
repair webpack build

Commit:
remove redundant code for production

Commit:
repair package.json script

Commit:
refactor

Commit:
refactor

Commit:
resolve map for children warning

Commit:
solve routing issue

Commit:
repair sign out redirect

Commit:
routing fix

Commit:
serve static files

Commit:
run npm start in procfile

Commit:
disable webpack hot module for development environment

Commit:
run production webpack

Commit:
repare jests module wrappers, add dependencies for sass

Commit:
add loader for sass

Commit:
addd scss stylesheet

Commit:
remove and rename files

Commit:
import scss style sheet

Commit:
rename file(feedback)

Commit:
refactor(rename files, repair dependencies)

Commit:
rename files, repair dependencies

Commit:
handle default login

Commit:
move login route to top

Commit:
remove production option from webpack build

Commit:
bug-fix(MessageStore.js): sort messages correctly using immutable-sorted

Commit:
bug-fix(package.json): install packages for immutable-sorted

Commit:
refactor(package.json): remove redundant dependency for immutable-sorted-map

Commit:
refactor(index.html): remove google scripts not needed

Commit:
refactor(index.scss): add style for modified google sign in button

Commit:
feature(firebaseFunctions.js): export google authentication provider

Commit:
refacotr(getMessagesAction.js): modify message action to remove duplication.

Commit:
refacotr(googleSignInAction): use pop up sign in with firebase authentication

Commit:
refactor(repair jsdocs): add appropriate parameter types

Commit:
refactor(group.js): remove file not needed

Commit:
add coverage files

Commit:
refactor(tests): remove redundant server tests

Commit:
refactor(tests): rename test files

Commit:
refactor(tests): rewrite test for firebase on frontend

Commit:
refactor(ActionTypes.js): correct recieve users constant

Commit:
refactor(read me): add contributions section

Commit:
refactor(actions): return promises and refactor actions

Commit:
refactor(utility): abstract get emails and numbers into seperate functions

Commit:
refactor(MessageListView.jsx): destructure props

Commit:
refactor(server utilities): remove files no more needed on server side

Commit:
refactor(send notifications): rename file using camel case

Commit:
refactor(.gitignore): add coverage files to git ignore

Commit:
bug(package.json): repair jest module wrapper for user stores

Commit:
refactor(request list): change import alias for request items

Commit:
refactor(server.js): refactor environment check test

Commit:
refactor(documentation): use appropriate jsdoc parameters, space out param values and return values

Commit:
refactor(index.scss): change message text color

Commit:
bug(resetPasswordAction): destructure properly

Commit:
refactor(remove coverage files from git cache)

Commit:
refactor(tests): change payload for reset password action test

Commit:
feature(reset password): add error message for non-existing email

Commit:
feature(reset password): handle error messages for reset password error

emmyakin commented on .babelrc:
```diff
@@ -1,3 +1,3 @@
-  "presets": ["es2015"]
-}
\ No newline at end of file
+  "presets": ["es2015", "env", "react"]
```
using env means you don't need to use es2015, it will be included by env

emmyakin commented on client/src/flux/Dispatcher.js:
```diff
@@ -0,0 +1,36 @@
+}
+
+const PostItDispatcher = new DispatcherClass();
+export default PostItDispatcher;
```
*nitpick*- you can just do `export default new DispatcherClass();` and omit line 34

emmyakin commented on client/src/flux/actions/acceptRejectGroupRequestAction.js:
```diff
@@ -0,0 +1,32 @@
+import PostItActionTypes from '../ActionTypes';
+import PostItDispatcher from '../Dispatcher';
+import { getAuth, getDatabase } from '../firebaseFunctions';
```
*nitpick* I like `firebaseHelpers` for module name

emmyakin commented on client/src/flux/actions/acceptRejectGroupRequestAction.js:
```diff
@@ -0,0 +1,32 @@
+  auth.onAuthStateChanged((user) => {
+    if (user) {
+      const userId = user.uid;
+      if (status === 'true') {
```
curious if status can be boolean rather than string

emmyakin commented on client/src/flux/actions/getAllUsersAction.js:
```diff
@@ -0,0 +1,62 @@
+      // get all users
+      const userReference = database.ref('/users/');
+
+      userReference.once('value', (snapshot) => {
```
this is just a side comment, but was wondering if you could store the group info for every user in the userref, so you don't have to get both snapshot and then run lodash difference, and you could possible just some sort of filter,
although that may lead to undesirable nesting too

emmyakin commented on client/src/flux/actions/getMessagesAction.js:
```diff
@@ -0,0 +1,62 @@
+
+      // get message Id from groups and iterate through messages node
+      messagesReference.orderByKey().on('child_added', (snapshot) => {
+        const messageKeys = [];
```
not sure this is needed, as  it is redeclared for every child_added event, and also, is it used anywhere asides from line 33.

emmyakin commented on client/src/flux/actions/getMessagesAction.js:
```diff
@@ -0,0 +1,62 @@
+      const messagesReference = database.ref(`/groups/${groupId}/messages`);
+
+      // notification value will be set if notification should happen
+      let notificationValue;
```
since this variable is modified, by each call of callback on line 35, no need to initialize it in this outside scope, you can initialize on line 35

emmyakin commented on client/src/flux/actions/getMessagesAction.js:
```diff
@@ -0,0 +1,62 @@
+        messageReference.orderByKey().on('value', (snap) => {
+          notificationValue = false;
+          const newMessage = snap.val();
+          newMessages.set(snap.key, snap.val());
```
replace snap.val() with newMessage

emmyakin commented on client/src/flux/actions/getMessagesAction.js:
```diff
@@ -0,0 +1,62 @@
+          notificationValue = false;
+          const newMessage = snap.val();
+          newMessages.set(snap.key, snap.val());
+          if (newMessage.read) {
```
line 39 -47 could be re-written declaratively I think as
```
const notify = (newMessage.read && !newMessage.read[username] && newMessage.sender !== username) ||
   (!newMessage.read && newMessage.sender !== username);

emmyakin commented on client/src/flux/actions/googleSignInAction.js:
```diff
@@ -0,0 +1,40 @@
+          PostItDispatcher.handleServerAction({
+            type: PostItActionTypes.FAILED_GOOGLE_LOGIN
+          });
+          return false;
```
remove, and add return to line 25

emmyakin commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,60 @@
+        loader: 'babel-loader',
+
+        query: {
+          presets: ['es2015']
```
no need to add presets if using .babelrc

emmyakin commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,60 @@
+        }
+      },
+      {
+        test: /\.(jsx)$/,
```
you can combine this with line 31 regex, and remove 40-43 `test: /\.jsx?$/,`

emmyakin commented on server/server.js:
```diff
@@ -1,46 +1,42 @@
-// Configure app
+const config= require('../webpack.config');   // eslint-disable-line
+const compiler = webpack(config);
+if (!process.env.development) {
```
use NODE_ENV here

emmyakin commented on server/controllerFunctions/sendNotifications.js:
```diff
@@ -0,0 +1,16 @@
+ * @return {void}
+ */
+export default function sendNotifications(req, res) {
+  const emails = req.body.emailList;
```
you can destructure here `const { emailList: emails, phoneNumbers: phones, priority } = req.body`

emmyakin commented on client/src/views/LoginRegisterContainer/RegisterComponent.jsx:
```diff
@@ -0,0 +1,172 @@
+   * @returns {void}
+   */
+  handleChange(event) {
+    const value = event.target.value;
```
`const { name, value } = event.target`

emmyakin commented on client/src/views/LoginRegisterContainer/LoginComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+   * @param {Object} event
+   */
+  handleChange(event) {
+    const value = event.target.value;
```
see below

emmyakin commented on client/src/views/GroupContainer/GroupList.jsx:
```diff
@@ -0,0 +1,138 @@
+    const status = MessageStore.getGroupNotificationDetails()
+      .status.get(newGroupWithNotification);
+
+    if (status) {
```
you can combine the ifs to `if (status && newGroupWithNotification)` by default empty string evaluates to false

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/userListView.jsx:
```diff
@@ -0,0 +1,70 @@
+   */
+  const showRequestMessageOrName = (key) => {
+    const statusMap = props.userStatus;
+    if (statusMap.get(key)) {
```
is it possible to just do `return statusMap.get(key)`

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/SendMessageView.jsx:
```diff
@@ -0,0 +1,157 @@
+
+    let level;
+
+    switch (newState) {
```
move switch logic to a separate function and return the values you need e.g.
```
switch (state) {
  case 0:
    return { level: 'gray', messagePriority = 'normal' };
  case 1:
 ...
}
```

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/SendMessageView.jsx:
```diff
@@ -0,0 +1,157 @@
+    event.preventDefault();
+    const priorityLevel = this.state.priorityLevel;
+    const newState = (priorityLevel + 1) % 3;
+    let messagePriority;
```
remove both declarations and use
`const { level, messagePriority } = getLevelAndPriority(newState);`

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/SendMessageView.jsx:
```diff
@@ -0,0 +1,157 @@
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
assign same object to defaultState maybe online 10, and use same object in here and on line 65

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/MessageListView.jsx:
```diff
@@ -0,0 +1,58 @@
+    <div className="message-list-items">
+      {messages && messages.valueSeq().map((message, key) => {
+        if (checkReadStatus(message, displayName)) {
+          return true;
```
what does react render when you `return true`, besides you may want to do a filter and then map

emmyakin commented on client/src/views/GroupContainer/GroupBodyContainer/AddUserView.jsx:
```diff
@@ -0,0 +1,91 @@
+   */
+  statusChange(index) {
+    let statusMap = this.state.userStatus;
+    statusMap = statusMap.set(index, true);
```
direct mutation here, do you really need a Map here, not sure, just asking

emmyakin commented on client/src/utility/getUsersEmailsNumbers.js:
```diff
@@ -0,0 +1,40 @@
+  return userIds.map(userId => (
+    new Promise((resolve) => {
+      // get user emails
+      const userEmailRef = db.ref(`users/${userId}/email`);
```
do you need to ref email and number, can you ref just the userId, and extract the email and number from there
you can then `resolve({ email, number});
meaning pass userIds as the only arg for the function

emmyakin commented on client/src/utility/getUsersEmailsNumbers.js:
```diff
@@ -0,0 +1,40 @@
+    });
+    const promises = getEmailNumbers(userIds, db, emails, numbers);
+    return Promise.all(promises)
+      .then(() => {
```
if the above comment applies, you can have .then(results)
results will be an array of emails and phone numbers

emmyakin commented on client/src/utility/getSenderInitials.js:
```diff
@@ -0,0 +1,13 @@
+// gets the initials of the message sender
+export default (sender) => {
+  let initials = '';
+  if (sender) {
```
would have love to use regex here, but could not figure it out

emmyakin commented on client/src/utility/getReadStatus.js:
```diff
@@ -0,0 +1,11 @@
+// get the read status for a message
+
+export default (message, displayName) => {
+  if (message.read) {
```
`return message.read && message.read[displayName]`

emmyakin commented on client/src/flux/stores/UserStore.js:
```diff
@@ -0,0 +1,102 @@
+
+
+  switch (action.type) {
+  case ActionTypes.LOGIN_USER:
```
indentation looks off

emmyakin commented on client/src/flux/stores/MessageStore.js:
```diff
@@ -0,0 +1,155 @@
+
+Dispatcher.register((payload) => {
+  const action = payload.action;
+  let groupId;
```
do you need all these variables defined here.
Can't you define in the message store constructor

emmyakin commented on client/src/flux/stores/MessageStore.js:
```diff
@@ -0,0 +1,155 @@
+
+const notificationMap = new Map();
+
+let groupWithNewMessageId;
```
avoid declaring variable outside of scope of functions or blocks that use them

emmyakin commented on client/src/flux/stores/GroupRequestStore.js:
```diff
@@ -0,0 +1,83 @@
+
+Dispatcher.register((payload) => {
+  const action = payload.action;
+  let groupId;
```
see comment below

emmyakin commented on client/src/flux/actions/registerUserAction.js:
```diff
@@ -0,0 +1,60 @@
+  const database = getDatabase();
+  const auth = getAuth();
+
+  if (validator.isEmail(email)) {
```
combine if statements whenever possible

Commit:
refactor(.babelrc): env preset takes care of es6 syntax

Commit:
refactor(Dispatcher.js): remove redundant variable declaration

Commit:
feedback(file name): change firebase functions to firebase helpers

Commit:
feedback(get messages action):

declare notification value in the callback scope since it is modified.

Commit:
feedback(get messages action): remove redundant message keys definition

Commit:
feedback(get messages action): use more declarative procedure for notifications.

Commit:
refactor(get initials): use regex to extract user intials

Commit:
refactor(send message): return dispatcher

Commit:
refactor(get read status): use succinct boolean expression

Commit:
feedback(send message action): use promise resolution to handle email and numbers

Commit:
feedback(send message view): abstract default state to new object

Commit:
feedback(webpack config): remove redundant preset for es6

Commit:
bug(fix email numbers retrieval)

Commit:
refactor(send notifications): use object shorthand

Commit:
bug(app.jsx): change import to firebase helpers

Commit:
feedback(GroupList.jsx): replace nested if with single complex boolean expression

Commit:
feedback(sendMessageView.js): abstract away priority level logic

Commit:
feedback(userListView.jsx): return boolean expression

Commit:
feedback(login and register): destructure event objects

Commit:
feedback(send notifications): destructure request body

Commit:
refactor(read me): use proper markdown

Commit:
Create LICENSE

Commit:
refactor(read me): add licensing information to read me

Commit:
refactor(tests): fix broken tests

Commit:
bug(token expiration): log out user if token is more than three days old

Commit:
fix(tests): fix broken test for  user store

Commit:
refactor(.eslintrc.json): repair indentation

Commit:
feedback(google sigin): add style, text to show google button action

Commit:
bug(error messages): solve zalgo problem of synchronous calls to error dispatch

Commit:
feedback(read me): add getting started information

Commit:
feedback(show loading): show loading button when getting groups

Commit:
refactor(js docs): space return and params

Commit:
feedback(show loading): show loading icon when still getting groups

Commit:
refactor(package.json): use production build configuration in webpack

Commit:
fix(webpack config): add loader for scss files

Commit:
add review app configuration for heroku

Commit:
fix(include sass loader): add sass loader to webpack configuration.

Commit:
fix(sass loader): remove sass loader temporarily

Commit:
add sass loader

Commit:
feedback(notifications): rename notifications files

Commit:
feedback(notifications): add appropriate jsdocs

Commit:
feedback(index.html): add author information and description to index.html

Commit:
fix(register action): validate username in register action

Commit:
feedback(group actions): abstract get request action to seperate function.

Commit:
feedback(user store): get token from user object, change login date identifier to camelcase

Commit:
feedback(header): pass sign out handler and username as props

Commit:
refactor(remove-preset): remove redundant preset from webpack config

[Finished]

Commit:
refactor(reword-readme): change wordings for contribute section

contribute session should read 'subscribes to airbnb styleguide'

[Finishes]

Commit:
refactor(consistent-tests): change jsx extensions

change .jsx file exensions in tests to .js to make tests consistent

[Finished]

Commit:
refactor(export-singleton): rename exported constants

`` rename constant in dispatcher to name of file ``

[Finished]

Commit:
refactor(export-singletons): change singletons to pascal case

`` singletons should be declared in pascal case and default export
should match file name for all stores ``

[Finished]

Commit:
refactor(seperate-actions): seperate group actions to different files

`` makes file structure consistent by seperating files in group actions
to different files ``

[Finished]

Commit:
style(mobile-responsive): make app mobile responsive

`` use the bootstrap cdn and media queries to cater for bigger screens
``

[Finished]

Commit:
refactor(fix-indentation): fix jsdoc indentation

[Finished]

Commit:
refactor(fix-jsdoc): add proper return type to jsdoc

[Finished]

Commit:
refactor(fix-jsdoc): use proper jsdoc return value

[Finished]

Commit:
refactor(remove-userId): remove user id reference in action

`` The user id is not needed in the action call, remove ``

[Finished]

Commit:
style(make-responsive): make component responsive

`` handle toggle state for mobile screen ``

[Finished]

Commit:
style(mobile-responsive): add bootstrap classes

`` use bootstrap classes to handle responsiveness for mobile screens ``

[Finished]

Commit:
style(style-clash): change class name of nav

`` avoid style clashes from bootstrap by changing class name ``

[Finished]

Commit:
refactor(remove-checkboxes): removes password checkboxes

[Finished]

Commit:
refactor(jsx-extension): remove jsx test extensions

`` remove the jsx extensions from some tests to make things consistent
``

[Finished]

Commit:
refactor(seperate-actions): seperate group actions

`` seperate group actions to different files to maintain consistency ``

[Finished]

Commit:
chore(add DS_Store): ignore DS_Store files

[Finished]

Commit:
fix(tests): fix and improve tests

 repair failing tests, increase coverage by covering more scenarios

[Finished]

Commit:
refactor(style-guide): name of export should match filename

change export to match the name of the file

[Finished]

fix(group-actions): remove group actions modulemapper

remove group actions module mapper, group actions no longer in a single
file

[Finished]

refactor(indentation): repair file indentation

[Finished]

Commit:
fix(token-extraction): extract token in store

use id token function to extract the id token in the store,
localstorage is populated in the callback of the function.

[Finished]

Commit:
fix(user-list): store user list correctly

user list should convert returned data to immutable-sorted map before
mergin to old list

[Finsihed]

Commit:
refactor(sign-in action): return user object alone

id token is retrieved in user store, send just user object back from
action to store

[Finished]

refactor(register-action): return user object alone

[Finished]

Commit:
refactor(google-sign): handle error for google sign in fail

google sign in fail should trigger a login error, sending payload to
error store to display error message for the user.

[Finished]

refactor(get-groups): handle error

send login error when request is made by an unauthenticated user

[Finished]

refactor(add-user): correct jsdoc description

[Finished]

Commit:
refactor(read me): show build status for feedback branch

[Finished]

Commit:
test(increase-coverage): test more scenarios

increase test coverage by testing behaviour of all action types

[Finished]

Commit:
test(increase-coverage): test helper functions

test all helper functions for correct behaviour

[Finished]

Commit:
documentatio(read me): point badges to feedback-implementation branch

[Finished]

Commit:
documentation(contribution-section): improve guidelines for
contributions

[Finished]

Commit:
fix(heroku-build): specify node version

specify node version for heroku build to avoid node-sass error

[Finished]

Commit:
chore(production-build): modify production build scripts

add build scripts that eliminate the need for babel-node

[Finished]

Commit:
chore(ignore dist): add server dist folder to gitignore

[Finished]

Commit:
chore(production-build): use dist folder for production build

[Finished]

Commit:
chore(task-runner): add concurrently dependency

[Finished]

Commit:
Merge pull request #73 from OsayamenOsaretin/chore/task-runner

Optimise production build

Commit:
style(send-message): fix padding for send message box

[Finished]

Commit:
Merge pull request #74 from OsayamenOsaretin/fix/send-message-padding

Fix send message padding

Commit:
tests(test-descriptions): use better test description

change test descriptions to describe exactly what is being tested for,
test more concretely

[Finished]

Commit:
Merge branch 'feedback-implementation' of https://github.com/OsayamenOsaretin/post-it-app into feedback-implementation
pull changes from remote repo

Commit:
chore(friendly-errors): add helper function for error handling

abstract away function to handle various firebase error types

[Finished]

Commit:
test(error-helper): add tests for friendly errors helper

[Finished]

Commit:
fix(add-group): fix bug with add-group feature

change auth state listener to plainly look for current user, so that
group doesn't get added for every auth state change

[Finished]

Commit:
fix(error-handling): fix error handling for register action

return appropriate error messages and use helper to return useful error
messages to the user

[Finished]

Commit:
fix(error-handling): fix error handling

use friendly error messages and cover more error cases

[Finished]

Commit:
refactor(username): extract username from localstorage

change username to state property and extract from localstorage in
constructor

[Finished]

Commit:
tests(fix-broken-tests): fix test

fix tests that broke due to improvements in error-handling and bug fixes

[Finished]

Commit:
fix(register-action): remove console messages

[Finished]

Commit:
tests(mock-dependency): use modified firebase mock

[Finished]

remove package-lock.json

Commit:
refactor(send-message): send message back in the result body

[Finished]

Commit:
refactor(export-server): export server

[Finished]

Commit:
refactor(remove-unused): remove unused code

[Finished]

Commit:
tests(add-mocks): add mocks for nexmo and nodemailer

manual mocks for nexmo and nodemailer

[Finished]

Commit:
tests(server-tests): add server side tests

[Finished]

Commit:
Merge pull request #77 from OsayamenOsaretin/test/server-side-test

Server Side Test

Commit:
fix(tests): fix failing test on travis

remove webpack requirement from test environment, no need for node
environment.

[Finished]

Commit:
tests(fix-travis-tests): ignore transpiled tests after build

target tests in the server/tests and igore tests in server/dist/tests

[Finished]

Commit:
tests(mock notifciationHelper): add mock for notification helper

no need to call notification helper in tests cases

[Finished]

Commit:
refactor(server): stop checking for test environment

since tests target server/__tests__ alone, no need to check for test
environment

[Finished]

Commit:
style(sender-initials): move sender intials closer to message body

[Finished]

Commit:
test(test-description): use better tests description

[Finished]

Commit:
tests(descriptions): use better test descriptions

[Finished]

Commit:
refactor(sourced-mocks): add disclaimer about origin for superagent mock

[Finished]

Commit:
refactor(mocks): remove unused mocks

[Finished]

Commit:
refactor(nitpicks): sort out spacing and remove redundant comments

[Finished]

Commit:
documentation(add jsdocs): add jsdoc for all functions, including
helpers

[Finished]

Commit:
Merge pull request #78 from OsayamenOsaretin/test/server-side-test

fix(tests): fix failing test on travis

Commit:
feedback(html-head): add tags for viewport

[Finished']

Commit:
Merge pull request #79 from OsayamenOsaretin/chore/update-html-head

Update HTML head

Commit:
style(refresh-spinner): fix loading groups spinner

[Finished]

Commit:
fix(groups-loading): add action type no group

[Start]

Commit:
tests(groups): add tests for loading

test loading feature in store and action tests

[Start]

Commit:
fix(groups-loading): handle case of no groups

terminate loading when no groups are found

[Finished]

Commit:
Merge pull request #80 from OsayamenOsaretin/chore/update-html-head

Fix spinner
